======================================================================================================================
DESCRIPTION:

tmf.ui: Move default to end of switch case

switch can contain a default clause for various reasons: to handle
unexpected values, to show that all the cases were properly considered.

For readability purpose, to help a developer to quickly find the default
behavior of a switch statement, it is recommended to put the default
clause at the end of the switch statement. This rule raises an issue if
the default clause is not the last one of the switch's cases.
Noncompliant Code Example

switch (param) {
  default: // default clause should be the last one
    error();
    break;
  case 0:
    doSomething();
    break;
  case 1:
    doSomethingElse();
    break;
}

Compliant Solution

switch (param) {
  case 0:
    doSomething();
    break;
  case 1:
    doSomethingElse();
    break;
  default:
    error();
    break;
}


Change-Id: I74d366abe502d94c19788f8204a915d6c7a93ef0
Signed-off-by: Matthew Khouzam <matthew.khouzam@ericsson.com>
Reviewed-on: https://git.eclipse.org/r/131774
Tested-by: CI Bot
Reviewed-by: Bernd Hufmann <bernd.hufmann@ericsson.com>
Tested-by: Bernd Hufmann <bernd.hufmann@ericsson.com>

======================================================================================================================
COMMENTS
======================================================================================================================
Author: Matthew Khouzam
Date: 2018-11-01 17:06:30.000000000
Message: 
Change has been successfully cherry-picked as 3a3ebd7ed448412de061c5c6d18ef597ae166813 by Matthew Khouzam
----------------------------------------------------------------------------------------------------------------------
