======================================================================================================================
DESCRIPTION:

tmf.ui: Move default to end of switch case

switch can contain a default clause for various reasons: to handle
unexpected values, to show that all the cases were properly considered.

For readability purpose, to help a developer to quickly find the default
behavior of a switch statement, it is recommended to put the default
clause at the end of the switch statement. This rule raises an issue if
the default clause is not the last one of the switch's cases.
Noncompliant Code Example

switch (param) {
  default: // default clause should be the last one
    error();
    break;
  case 0:
    doSomething();
    break;
  case 1:
    doSomethingElse();
    break;
}

Compliant Solution

switch (param) {
  case 0:
    doSomething();
    break;
  case 1:
    doSomethingElse();
    break;
  default:
    error();
    break;
}


Change-Id: I74d366abe502d94c19788f8204a915d6c7a93ef0
Signed-off-by: Matthew Khouzam <matthew.khouzam@ericsson.com>
======================================================================================================================
COMMENTS
======================================================================================================================
Author: Matthew Khouzam
Date: 2018-10-31 20:12:09.000000000
Message: 
Uploaded patch set 1.
----------------------------------------------------------------------------------------------------------------------
Author: CI Bot
Date: 2018-10-31 20:49:56.000000000
Message: 
Patch Set 1:

Build Started https://hudson.eclipse.org/tracecompass/job/tracecompass-gerrit/20145/
----------------------------------------------------------------------------------------------------------------------
Author: CI Bot
Date: 2018-10-31 21:16:58.000000000
Message: 
Patch Set 1: Verified-1

Build Failed 

https://hudson.eclipse.org/tracecompass/job/tracecompass-gerrit/20145/ : FAILURE
----------------------------------------------------------------------------------------------------------------------
