======================================================================================================================
DESCRIPTION:

Bug 513614 - Remove invalid e4 add-ons and handlers

Search for descriptors, handlers and addons in different threads in
order to remove invalid application model elements

Change-Id: I9c50959729bc09f38ffad266c10e41e6890f9040
Signed-off-by: Simon Scholz <simon.scholz@vogella.com>
======================================================================================================================
COMMENTS
======================================================================================================================
Author: Lars Vogel
Date: 2018-06-21 09:40:35.000000000
Message: 
Patch Set 3: Patch Set 2 was rebased
----------------------------------------------------------------------------------------------------------------------
Author: CI Bot
Date: 2018-06-21 10:22:21.000000000
Message: 
Patch Set 3:

Build Started https://ci.eclipse.org/platform/job/eclipse.platform.ui-Gerrit/15249/
----------------------------------------------------------------------------------------------------------------------
Author: CI Bot
Date: 2018-06-21 11:07:19.000000000
Message: 
Patch Set 3: Verified+1

Build Successful 

https://ci.eclipse.org/platform/job/eclipse.platform.ui-Gerrit/15249/ : SUCCESS
----------------------------------------------------------------------------------------------------------------------
Author: Lars Vogel
Date: 2018-06-21 13:49:34.000000000
Message: 
Patch Set 3:

Could you add a few unit tests for this? org.eclipse.e4.ui.tests would be IMHO a good place to place it. It should also contain similar tests.
----------------------------------------------------------------------------------------------------------------------
Author: Lars Vogel
Date: 2018-06-21 14:21:12.000000000
Message: 
Patch Set 3:

(4 comments)

I suggest you start using SonarLint, all my feedback comes from SonarLint ;-)
Line:68, bundles/org.eclipse.ui.ide.application/src/org/eclipse/ui/internal/ide/application/addons/ModelCleanupAddon.java -> Please add final

Line:69, bundles/org.eclipse.ui.ide.application/src/org/eclipse/ui/internal/ide/application/addons/ModelCleanupAddon.java -> final

Line:173, bundles/org.eclipse.ui.ide.application/src/org/eclipse/ui/internal/ide/application/addons/ModelCleanupAddon.java -> contributionURI can be null here

Line:195, bundles/org.eclipse.ui.ide.application/src/org/eclipse/ui/internal/ide/application/addons/ModelCleanupAddon.java -> Return of boolean literal statements wrapped into if-then-else ones should be simplified.

Noncompliant Code Example

if (expression) {
  return true;
} else {
  return false;
}
Compliant Solution

return expression;

----------------------------------------------------------------------------------------------------------------------
